src/main/java/forestry/core/utils/MathUtil.java

```diff
@@ -0,0 +1,29 @@
+package forestry.core.utils;
+
+public class MathUtil {
+
+    public static float safeMultiply(float a, float b) {
+        if (a > 0 && b > 0 && a > Float.MAX_VALUE / b) {
+            return Float.MAX_VALUE;
+        } else if (a > 0 && b < 0 && a > Float.MIN_VALUE / b) {
+            return Float.MIN_VALUE;
+        } else if (a < 0 && b > 0 && a < Float.MIN_VALUE / b) {
+            return Float.MIN_VALUE;
+        } else if (a < 0 && b < 0 && a < Float.MAX_VALUE / b) {
+            return Float.MAX_VALUE;
+        } else {
```

Comment:
* `Float.MIN_VALUE` ***does not*** represent the biggest negative number; it represents smallest positive number (=0.00....01).
* All these branches can be simplified using `Math.abs` and `Math.signum`.

The same issues are valid for `safeAdd`.

Reply:
@Connor-Colenso

Reply:
Rewrote it using bigdecimal, the logic is imo easier to follow, even if it's a little slower.